jasmine/spec/inverted-index.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @class InvertedIndex
+ */
+class InvertedIndex {
+  /**
+   * @constructor
+   */
+  constructor() {
+    this.files = {};
+    this.indexMap = {};
+  }
+
+  /**
+   * createIndex
+   * @param {Object} fileName name of file to map index for
+   * @return {Boolean} false if index is not created
+   */
+  createIndex(fileName) {
+    if (Object.hasOwnProperty.call(this.files, fileName)) {
+      const currentFileContent = this.files[fileName];
+      this.indexMap[fileName] = this.indexMap[fileName] || {};
+      const fileLength = this.files[fileName].length;
+      for (let docIndex = 0; docIndex < fileLength; docIndex += 1) {
+        const currentFileDoc = currentFileContent[docIndex];
+        const docTitle = currentFileDoc.title;
+        const docText = currentFileDoc.text;
+        const docToken = this.getCleanTokens(`${docText} ${docTitle}`);
+        docToken.forEach((word) => {
+          if (word in this.indexMap[fileName]) {
+            if (this.indexMap[fileName][word].indexOf(docIndex) === -1) {
+              this.indexMap[fileName][word].push(docIndex);
+            }
+          } else {
+            this.indexMap[fileName][word] = [docIndex];
+          }
+        });
+      }
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * getIndex
+   * Returns index map of a file
+   * @param {String} fileName name of file to return index map
+   * @return {Object} a key pair value of file index map
+   */
+  getIndex(fileName) {
+    return this.indexMap[fileName];
+  }
+
+  /**
+   * searchIndex
+   * Search for the occurrence of words in the indexMap
+   * @param {String} searchTerms The search term(s)
+   * @param {Object} fileNames An array of filenames to search
+   * @returns {Object} A map of the search result
+   */
+  searchIndex(searchTerms, fileNames) {
```

Comment:
Thank you @andela-cnwankwo.

The issue seems to be a browser thing and I will be looking into it.